File: src/main/webapp/js/service/LinkService.js
//服务层
app.service('linkService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../link/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../link/findPage.do?pageNum='+page+'&pageSize='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../link/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../link/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../link/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../link/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../link/search.do?pageNum='+page+"&pageSize="+rows, searchEntity);
	}    	
});

File: src/main/java/com/wyan/service/LinkService.java
package com.wyan.service;

import com.github.pagehelper.PageInfo;
import com.wyan.pojo.TbLink;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface LinkService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbLink> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageInfo findPage(int pageNum, int pageSize);


	/**
	 * 增加
	*/
	public void add(TbLink link);


	/**
	 * 修改
	 */
	public void update(TbLink link);


	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbLink findOne(Integer linkid);


	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Integer[] linkids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageInfo findPage(TbLink link, int pageNum, int pageSize);
	
}
